src/qibo/models/circuit.py

```diff
@@ -1207,191 +1216,8 @@ def from_qasm(cls, qasm_code, accelerators=None, density_matrix=False):
                 c2.add(gates.H(1))
                 c2.add(gates.CNOT(0, 1))
         """
-        nqubits, gate_list = cls._parse_qasm(qasm_code)
-        circuit = cls(nqubits, accelerators, density_matrix)
-        for gate, qubits, params in gate_list:
-            if gate == gates.M:
-                circuit.add(gate(*qubits, register_name=params))
-            elif params is None:
-                circuit.add(gate(*qubits))
-            else:
-                # assume parametrized gate
-                circuit.add(gate(*qubits, *params))
-        return circuit
-
-    @staticmethod
-    def _parse_qasm(qasm_code: str):
-        """Extracts circuit information from QASM script.
-
-        Helper method for ``from_qasm``.
-
-        Args:
-            qasm_code: String with the QASM code to parse.
-
-        Returns:
-            nqubits: The total number of qubits in the circuit.
-            gate_list: List that specifies the gates of the circuit.
-                Contains tuples of the form
-                (Qibo gate name, qubit IDs, optional additional parameter).
-                The additional parameter is the ``register_name`` for
-                measurement gates or ``theta`` for parametrized gates.
-        """
-        import re
-
-        def read_args(args):
-            _args = iter(re.split(r"[\[\],]", args))
-            for name in _args:
-                if name:
-                    index = next(_args)
-                    if not index.isdigit():
-                        raise_error(ValueError, "Invalid QASM qubit arguments:", args)
```

Comment:
Apparently this is one of two lines in `qibo` which the `raise_error` function was ran by being called as `raise_error(exception, message, args)`. Removing these two lines is what is preventing 100% coverage